frontend/src/charts/trendsChart/Axes.tsx

```diff
@@ -0,0 +1,173 @@
+/**
+ * Axes for the charts that track trends over time
+ * Uses d3.js to apply generate and draw axes on an SVG
+ * @param {object[]} data array of timeseries data objects
+ * @param {*} xScale a d3 time series scale function
+ * @param {*} yScale a d3 linear scale function
+ * @param {number} width the width of the svg
+ * @param {number} marginBottom the margin below the line chart (dynamic for mobile & desktop)
+ * @param {number} marginLeft the margin to the left of the line chart
+ * @param {number} marginRight the margin to the right of the line chart
+ * @param {object} axisConfig an object containing the configuration for axes - type and labels
+ * @param {boolean} isMobile a flag to determine whether user is viewing app below the mobile breakpoint
+ * returns jsx of an svg group containing groups of axes and axis labels
+
+/* External Imports */
+import React, { useRef, useEffect } from "react";
+import { axisLeft, axisBottom, select } from "d3";
+
+/* Local Imports */
+
+/* Styles */
+import styles from "./Trends.module.scss";
+
+/* Constants */
+import { CONFIG, TYPES, FORMATTERS as F } from "./constants";
+import { TrendsData, XScale, YScale, AxisConfig } from "./types";
+
+/* Helpers */
+import { getMinNumber, getMaxNumber } from "./helpers";
+/* Define type interface */
+export interface AxesProps {
+  data: TrendsData;
+  xScale: XScale;
+  yScale: YScale;
+  width: number;
+  marginBottom: number;
+  marginLeft: number;
+  marginRight: number;
+  axisConfig: AxisConfig;
+  isMobile: boolean;
+}
+
+/* Render component */
+export function Axes({
+  data,
+  xScale,
+  yScale,
+  width,
+  marginBottom,
+  marginLeft,
+  marginRight,
+  axisConfig,
+  isMobile,
+}: AxesProps) {
+  /* Config */
+  const { HEIGHT, TICK_PADDING, Y_AXIS_LABEL_PADDING, MOBILE } = CONFIG;
+  const { type, yAxisLabel = "" } = axisConfig || {};
+  const yAxisLabelPadding = isMobile
+    ? MOBILE.Y_AXIS_LABEL_PADDING
+    : Y_AXIS_LABEL_PADDING;
+  // handles difference between per100k and percent_share charts
+  const Y_AXIS_CONFIG = {
+    [TYPES.HUNDRED_K]: {
+      topLabel: F.capitalize(yAxisLabel) + " →", // reference to shortLabel from metricConfig
+      bottomLabel: "",
+      formatter: (d: string | number) => d,
+    },
+    [TYPES.PERCENT_SHARE]: {
+      // @ts-ignore
+      topLabel: getMaxNumber(data) <= 0 ? "" : "Over-represented →",
+      // @ts-ignore
```

Comment:
Just curious; what didn't typescript like about this?

Reply:
getMaxNumber(data) could be undefined. I fixed it with an || statement